From bd4334664ed1069ad491b3baada6fc0128eb1029 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Sat, 12 Jan 2019 01:11:08 +0800
Subject: [PATCH 44/54] Input: vmmouse: change wheel capability to the abs
 device

This makes vmmouse be classified as a regular mouse pointer
in Android.

Change-Id: I8792caf43a563d2a67d833b6b90565fd068ecabf
---
 drivers/input/mouse/vmmouse.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/input/mouse/vmmouse.c b/drivers/input/mouse/vmmouse.c
index 42443ffba7c4..73456f7696b9 100644
--- a/drivers/input/mouse/vmmouse.c
+++ b/drivers/input/mouse/vmmouse.c
@@ -192,7 +192,7 @@ static psmouse_ret_t vmmouse_report_events(struct psmouse *psmouse)
 		}
 
 		/* Xorg seems to ignore wheel events on absolute devices */
-		input_report_rel(rel_dev, REL_WHEEL, -(s8)((u8) z));
+		input_report_rel(abs_dev, REL_WHEEL, -(s8)((u8) z));
 
 		vmmouse_report_button(psmouse, abs_dev, rel_dev,
 				      pref_dev, BTN_LEFT,
@@ -420,7 +420,7 @@ static int vmmouse_reconnect(struct psmouse *psmouse)
 int vmmouse_init(struct psmouse *psmouse)
 {
 	struct vmmouse_data *priv;
-	struct input_dev *rel_dev = psmouse->dev, *abs_dev;
+	struct input_dev *abs_dev;
 	int error;
 
 	psmouse_reset(psmouse);
@@ -459,6 +459,7 @@ int vmmouse_init(struct psmouse *psmouse)
 	input_set_capability(abs_dev, EV_KEY, BTN_MIDDLE);
 	input_set_capability(abs_dev, EV_ABS, ABS_X);
 	input_set_capability(abs_dev, EV_ABS, ABS_Y);
+	input_set_capability(abs_dev, EV_REL, REL_WHEEL);
 	input_set_abs_params(abs_dev, ABS_X, 0, VMMOUSE_MAX_X, 0, 0);
 	input_set_abs_params(abs_dev, ABS_Y, 0, VMMOUSE_MAX_Y, 0, 0);
 
@@ -466,9 +467,6 @@ int vmmouse_init(struct psmouse *psmouse)
 	if (error)
 		goto init_fail;
 
-	/* Add wheel capability to the relative device */
-	input_set_capability(rel_dev, EV_REL, REL_WHEEL);
-
 	psmouse->protocol_handler = vmmouse_process_byte;
 	psmouse->disconnect = vmmouse_disconnect;
 	psmouse->reconnect = vmmouse_reconnect;
-- 
2.34.1

